fe/fe-core/src/main/java/org/apache/doris/planner/external/MaxComputeScanNode.java

```diff
@@ -74,7 +78,37 @@ protected Map<String, String> getLocationProperties() throws UserException {
     @Override
     protected List<Split> getSplits() throws UserException {
         List<Split> result = new ArrayList<>();
-        result.add(new FileSplit(new Path("/"), 0, -1, -1, 0L, new String[0], Collections.emptyList()));
+        // String splitPath = catalog.getTunnelUrl();
+        // TODO: use single max compute scan node rather than file scan node
+        com.aliyun.odps.Table odpsTable = table.getOdpsTable();
+        if (desc.getSlots().isEmpty() || odpsTable.getFileNum() <= 0) {
+            return result;
+        }
+        try {
+            List<Pair<Long, Long>> sliceRange = new ArrayList<>();
+            long totalRows = catalog.getTotalRows(table.getDbName(), table.getName());
+            long fileNum = odpsTable.getFileNum();
+            long splitSize = (long) Math.ceil((double) totalRows / fileNum);
```

Comment:
If the `totalRows` is less than `fileNum`, the `splitSize` will be 0.

Reply:
And what if there is only 100 rows, and 10 files. Do we need to split into 10 ranges? I think this is too small.